addons/web/static/tests/views/form_tests.js

```diff
@@ -9853,6 +9853,65 @@ QUnit.module('Views', {
         actionManager.destroy();
     });
 
+    QUnit.test('Auto save: save on closing tab/browser', async function (assert) {
+        assert.expect(2);
+
+        const form = await createView({
+            View: FormView,
+            model: 'partner',
+            data: this.data,
+            arch: `
+                <form>
+                    <group>
+                        <field name="display_name"/>
+                    </group>
+                </form>`,
+            res_id: 1,
+        });
+
+        await testUtils.form.clickEdit(form);
+        assert.notStrictEqual(form.$('.o_field_widget[name="display_name"]').val(), 'test');
+
+        await testUtils.fields.editInput(form.$('.o_field_widget[name="display_name"]'), 'test');
+        window.dispatchEvent(new Event("beforeunload"));
+        await testUtils.nextTick();
```

Comment:
mockRPC to check that it has been saved? Typically, after the beforeunload there is nothing left to do as the tab is supposed to be closed :-)

Reply:
also assert that the event isn't prevented